internal/git/gitsync_processor.go

```diff
@@ -122,6 +121,7 @@ func watchRepo(ctx context.Context, r *git.Repository, gitSync *v1.GitSync, rest
 		}
 	}
 	val, ok := gitSync.Status.CommitStatus[repo.Name]
+	lastCommitHash = gitSync.Status.CommitStatus[repo.Name].Hash
```

Comment:
this should only be in the case of `ok==true`

Reply:
and in the case of `ok==false`, we still need to set it there after we apply the resources (to the hash)